app/scripts/athena_partition_archive.py

```diff
@@ -325,71 +498,34 @@ def main():
                 s3_location = row["s3_location"]
-
-                LOGGER.info('Processing %s.%s', database_name, table_name)
-
-                sql = "show partitions " + database_name + "." + table_name
-
-                response = execute_athena(sql, database_name)
-                query_file = response['QueryExecution']['QueryExecutionId'] + '.txt'
-
-                # Sleep to allow file to be written to S3
-                time.sleep(1)
-
-                S3.download_file(ATHENA_LOG, query_file, "/APP/query.txt")
-
-                result_list = []
-
-                with open("/APP/query.txt") as file:
-                    for line in file:
-                        line = line.strip()
-                        result_list.append(line)
-
-                partition_list = []
-
-                for item in result_list:
-                    try:
-                        match = PATTERN.search(item).group(0)
-                    except:
-                        LOGGER.info("No match found.")
-                    if match <= str(MAXCLEARDOWN):
-                        partition_list.append(item)
-
-                for item in partition_list:
-                    item_quoted = item[:10] + "'" + item[10:] + "'"
-                    item_stripped = item.split('=')[1]
-
-                    drop_partition_sql = ("ALTER TABLE " + database_name + "." + table_name + \
-                                         " DROP PARTITION (" + item_quoted + ");")
-                    add_partition_sql = ("ALTER TABLE " + database_name + "." + table_name + \
-                                         "_archive ADD PARTITION (" + item_quoted + ") LOCATION 's3://" + s3_location + "/" + item_stripped + "';")
-
-
-                    try:
-                        LOGGER.info('Dropping partition "%s" from "%s.%s"', item, database_name, table_name)
-                        LOGGER.debug(drop_partition_sql)
-                        execute_athena(drop_partition_sql, database_name)
-                    except Exception as err:
-                        send_message_to_slack(err)
-                        error_handler(sys.exc_info()[2].tb_lineno, err)
-                        sys.exit(1)
-
-                    try:
-                        LOGGER.info('Adding partition "%s" from "%s.%s"', item, database_name, table_name)
-                        LOGGER.debug(add_partition_sql)
-                        execute_athena(add_partition_sql, database_name)
-                    except Exception as err:
-                        send_message_to_slack(err)
-                        error_handler(sys.exc_info()[2].tb_lineno, err)
-                        sys.exit(1)
-
-                LOGGER.info("Complete.")
+                retention_period = row["retention_period"]
+
+                origin_table = check_table(database_name, table_name)
+                if origin_table:
+
+                    archive_table = check_table(database_name, table_name)
```

Comment:
Looks like table_name should be different to above check.

Reply:
Correctly specified archive table name.